GET on subscribe form

meejah opened the pull request:
This doesn't email support if e.g. GoogleBot visits our form with a GET instead of POST. Also, fix dependencies.

vu3rdd commented:
Looks good to me.

exarkun commented on requirements.txt:
```diff
@@ -1,10 +1,11 @@
 twisted
-
-https://tahoe-lafs.org/source/tahoe-lafs/deps/tahoe-dep-sdists/txAWS-0.2.1.post5.tar.gz
+https://tahoe-lafs.org/source/tahoe-lafs/deps/tahoe-lafs-dep-sdists/txAWS-0.2.1.post5.tar.gz
```
Ah, also, I noticed some pip-invoking code that does `--find-links https://tahoe-lafs.org/deps/`.  That page includes the correct link to txAWS-0.2.1.post5.  I wonder if someone has the intention that the full URL of this archive be considered an implementation detail.  If that's the case, this line should really just be `txAWS` (or `txAWS==0.2.1.post5`?) and the correct `--find-links` argument should also be used.

exarkun commented on lae_site/handlers/submit_subscription.py:
```diff
@@ -137,6 +137,12 @@ def render(self, request):
+        if request.method != 'POST':
+            tmpl = env.get_template('s4-subscription-form.html')
+            msg = ("Your browser requested this page with a GET request but we expected a POST."
+                   "\n\nPlease try again.")
```
I see there is some test coverage for this code.  It leans very heavily on mocks and is pretty light on documentation.  The consequences of which are probably that the tests present a high barrier to maintenance and that they may only catch the most superficial of problems.  Given that, I can see the pressure to not bother updating them.

Perhaps this simple change would be a good opportunity to replace the existing tests with some new, more valuable ones.

Note that the existing tests fail with this change because they use a very primitive test double of `Request` which has no `method` attribute (this is an example of how the current tests present a high barrier to maintenance).

One way or another, it seems like the failing tests should be considered a blocker to merging this change, though.  As far as I can tell, the test suite is green on master@HEAD and red for this branch.

meejah commented:
Okay, I think I addressed @exarkun's review, and added tests for this case -- I agree the tests seem a little unwieldy *and* that they should still all run ;)

p.s. don't know how you feel about force-pushes, but I tend to do that to keep all the discussion in the same PR (e.g. instead of opening another one) as I re-based this to master and changed the commits a little.
